Packs/CVE_2025_31324/Playbooks/playbook-CVE-2025-31324_-_SAP_NetWeaver_Visual_Composer.yml

```diff
@@ -0,0 +1,2835 @@
ook should be triggered manually or can be configured as a job.\nPlease create a new incident and choose the CVE-2025-31324 - SAP NetWeaver Visual Composer playbook and Rapid Breach Response incident type.\n\nCVE-2025-31324 is a critical zero-day vulnerability affecting the Metadata Uploader component of SAP NetWeaver Visual Composer. The vulnerability arises from missing authorization checks, allowing unauthenticated attackers to upload malicious executable binaries. Exploitation of this flaw can lead to full remote code execution (RCE) on affected systems, posing a significant risk to confidentiality, integrity, and availability.\n\n## CVE-2025-31324 - SAP NetWeaver RCE Vulnerability\n\n##   Vulnerability Overview\n\n- **Component Affected**: SAP NetWeaver Visual Composer Metadata Uploader  \n- **Endpoint**: `/developmentserver/metadatauploader`  \n- **CVE ID**: CVE-2025-31324  \n- **CVSS Score**: 10.0 (Critical)  \n- **Exploitability**: Unauthenticated remote attackers can exploit this without user interaction  \n\nThis flaw allows unauthenticated attackers to upload arbitrary files (e.g., JSP web shells), enabling remote code execution with the same privileges as the SAP application server process.  \n[Source: Unit42 - Palo Alto Networks](https://unit42.paloaltonetworks.com/threat-brief-sap-netweaver-cve-2025-31324/)\n\n\n##  Mitigation and Recommendations\n\n- **Apply Patch**: SAP Note #3594142 (released April 24, 2025)  \n- **Disable Visual Composer** if not in use  \n- **Restrict Access** to the vulnerable endpoint  \n- **Monitor for IoCs** in `/irj/servlet_jsp/irj/root/` and suspicious traffic  \n\n##  Conclusion\n\nCVE-2025-31324 is actively exploited and is critically severe. Organizations should patch immediately, monitor for compromise, and disable or restrict vulnerable components.\n\n[View official CVE details on NIST](https://nvd.nist.gov/vuln/detail/CVE-2025-31324)\n\n##  Playbook Triggers\n- Manually\n- \"CVE Exploitation - 986328356\" Agent rule\n\n##  Playbook Flow\n1. Using XQL, identify potential SAP NetWeaver instances in your environment.\n2. Using XQL, check if there are events that point to any potential webshells downloaded in the directories.\n3. Directory enumeration to identify if there are already any suspicious files that might indicate a webshell.\n4. *Collect* IOCs from a Unit42 blog, *search* them using the \"Threat Hunting - Generic\" playbook (supports Palo Alto networks products, Qradar, and Splunk), and *block* those indicators using the \"Containment Plan - Block Indicators\" playbook.\n5. Download Sigma rules.\n6. Provides Mitigation recommendations.\n\nNote: This is a beta playbook, which lets you implement and test pre-release software. Since the playbook is beta, it might contain bugs. Updates to the pack during the beta phase might include non-backward compatible features. We appreciate your feedback on the quality and usability of the pack to help us identify issues, fix them, and continually improve."
```

Comment:
2 notes:
- change the containmnet plan  --> to `Block Indicators - Generic v3`
- Should we reorder the steps per the playbook flow?